include/wlr/types/wlr_keyboard.h

```diff
@@ -67,6 +69,8 @@ struct wlr_keyboard {
 		int32_t delay;
 	} repeat_info;
 
+	struct wlr_virtual_keyboard_v1 *virtual_keyboard;
```

Comment:
Instead of adding a new field, we should add a way to get a `wlr_virtual_keyboard_v1 directly from a `wlr_keyboard` or a `wlr_input_device`. Ideally this should be done in a standalone commit. See e.g. `get_libinput_device_from_device`.

Reply:
I have add a `wlr_input_device_get_virtual_keyboard` to get from `wlr_input_device`.